src/FileDownloader.ts

```diff
@@ -4,10 +4,40 @@
 import { ExtensionContext, CancellationToken, Uri } from "vscode";
 
 export interface FileDownloadSettings {
+    /**
+     * Timeout in milliseconds for the request.
+     * @default 5000
+     */
     timeoutInMs?: number;
+    /**
+     * Number of retries for the request.
+     * @default 5
+     */
     retries?: number;
```

Comment:
☕️🎊💡Curious question:  Just wondering the the use of `nullable` type for `retries` which could be `0` to start with, but I am missing built up context.

Reply:
I'm missing the context too - the only thing I can think of is this is the interface, so not really it's responsibility to define the defaults of the actual implementation. I could remove the defaults from the interface documentation?